lib/plugin.js

```diff
@@ -14,8 +15,9 @@ function plugin (pluginOptions) {
     constructSignatureString,
     getDigest,
     digestEncoding: 'base64',
-    getAlgorithm: () => {
-      throw new Error('No getAlgorithm function provided.')
+    getAlgorithm,
+    algorithmMap: () => {
```

Comment:
I don't think we need a default like this for now, this is also mismatched from the object type we expect the user to provide. I would like to validate these options after we construct the options object. We can iterate over the keys and ensure everything we expect exists and is the correct type.